parser/preinstrumentationtracing_test.go

```diff
@@ -0,0 +1,169 @@
+		newrelic.ConfigAppName("Short Lived App"),
+		newrelic.ConfigLicense(os.Getenv("NEW_RELIC_LICENSE_KEY")),
+		newrelic.ConfigDebugLogger(os.Stdout),
+	)
+	if nil != err {
+		fmt.Println(err)
+		os.Exit(1)
+	}
+
+	tasks := []string{"white", "black", "red", "blue", "green", "yellow"}
+	for _, task := range tasks {
+		txn := app.StartTransaction("task")
+		time.Sleep(10 * time.Millisecond)
+		fmt.Println("Task:", task)
+		txn.End()
+		app.RecordCustomEvent("task", map[string]interface{}{
+			"color": task,
+		})
+	}
+	app.Shutdown(10 * time.Second)
+}
+`,
+
+			expect: map[string][]string{
+				"txn": {"Sleep", "Println", "txn.End"},
+			},
+		},
+		{
+			name: "Detect Transactions passed through function",
+			code: `package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+func processTask(txn *newrelic.Transaction, task string) {
+	time.Sleep(10 * time.Millisecond)
+	fmt.Println("Processing task:", task)
+	txn.End()
+}
+
+func main() {
+	app, err := newrelic.NewApplication(
+		newrelic.ConfigAppName("Short Lived App"),
+		newrelic.ConfigLicense(os.Getenv("NEW_RELIC_LICENSE_KEY")),
+		newrelic.ConfigDebugLogger(os.Stdout),
+	)
+	if nil != err {
+		fmt.Println(err)
+		return
+	}
+
+	tasks := []string{"task1", "task2", "task3"}
+	for _, task := range tasks {
+		txn := app.StartTransaction("task")
+		processTask(txn, task)
+	}
+	app.Shutdown(10 * time.Second)
+}`,
+
+			expect: map[string][]string{
+				"txn": {"processTask", "Sleep", "Println", "txn.End"},
+			},
+		},
+		{
+			name: "Logic after transaction ends should not be captured",
+			code: `package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+func main() {
+	app, err := newrelic.NewApplication(
+		newrelic.ConfigAppName("Short Lived App"),
+		newrelic.ConfigLicense(os.Getenv("NEW_RELIC_LICENSE_KEY")),
+		newrelic.ConfigDebugLogger(os.Stdout),
+	)
+	if err != nil {
+		fmt.Println(err)
+		return
+	}
+
+	txn := app.StartTransaction("task")
+	fmt.Println("Starting task")
+	txn.End()
+
+	fmt.Println("Task ended. Performing cleanup.") // This should not be part of the transaction expressions
+	time.Sleep(5 * time.Second)
+	app.Shutdown(10 * time.Second)
+}`,
+			expect: map[string][]string{
+				"txn": {"Println", "txn.End"},
+			},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			defer panicRecovery(t)
+			id, err := pseudo_uuid()
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			testDir := fmt.Sprintf("tmp_%s", id)
+			defer cleanTestApp(t, testDir)
+
+			manager := testInstrumentationManager(t, tt.code, testDir)
+			pkg := manager.getDecoratorPackage()
+			if pkg == nil {
+				t.Fatalf("Package was nil: %+v", manager.packages)
+			}
+
+			manager.tracingFunctions.preinstrumentation = append(manager.tracingFunctions.preinstrumentation, DetectTransactions)
+
+			manager.loadPreInstrumentationTracingFunctions(DetectTransactions)
+			err = manager.ScanApplication(DetectTransactions)
```

Comment:
The argument here is unnecessary since the `DetectTransactions` function was already registered above. We can probably remove this (and the need for an arg to this `ScanApplication` function altogether)
```suggestion
			err = manager.ScanApplication()
```